api/desecapi/tests/test_pdns_change_tracker.py

```diff
@@ -0,0 +1,461 @@
+                      '2001:bb5e:921:b17f:7c9b:afb6:9933:cc79', '2001:a861:7139:e21e:11e4:8782:242b:e2a2',
+                      '2001:eaa:ff53:c819:93e:437c:ccc8:330c', '2001:6a88:fb92:5b43:984b:b729:393b:f173']
+    ALT_CONTENT_VALUES = ['2001:2d03:6247:3494:b92e:d4a:2827:e2d', '2001:4b37:19d6:b66e:1aa1:db0f:98b5:d065',
+                          '2001:dbf1:e401:ace2:bc99:eb22:6e12:ec81', '2001:fa92:3564:7c3f:9995:2068:58bf:2a45',
+                          '2001:4c2c:c671:9f0c:600e:4eb6:672e:48c7', '2001:5d09:a6f7:594b:afa4:318a:6eda:3ec6',
+                          '2001:f33a:407c:f4e6:f886:dce2:6d08:d8ae', '2001:43c8:378d:7d37:92eb:fb0c:26b1:4998',
+                          '2001:7293:88c5:5405:fd1:7334:bb55:be20', '2001:c4b7:ae76:a9a2:ffb5:ba30:6874:a416',
+                          '2001:175f:7880:ef82:b65a:a472:14c9:a495', '2001:8c35:1566:4f53:c26a:c54:2c9f:1463']
+
+
+class TXTRRTestCase(RRTestCase):
+    SUBNAME = '_acme_challenge'
+    TYPE = 'TXT'
+    TTL = 876
+    CONTENT_VALUES = ['The quick brown fox jumps over the lazy dog',
+                      'main( ) {printf("hello, world\n");}',
+                      '“红色联合”对“四·二八兵团”总部大楼的攻击已持续了两天"']
+    ALT_CONTENT_VALUES = ['🧥 👚 👕 👖 👔 👗 👙 👘 👠 👡 👢 👞 👟 🥾 🥿 🧦 🧤 🧣 🎩 🧢 👒 🎓 ⛑ 👑 👝 👛 👜 💼 🎒 👓 🕶 🥽 🥼 🌂 🧵',
+                          'v=spf1 ip4:192.0.2.0/24 ip4:198.51.100.123 a -all',
+                          'https://en.wikipedia.org/wiki/Domain_Name_System']
+
+
+class RRSetTestCase(PdnsChangeTrackerTestCase):
+    TEST_DATA = {
+        ('A', '_asdf', 123): ['1.2.3.4', '5.5.5.5'],
+        ('TXT', 'test', 455): ['ASDF', 'foobar', '92847'],
+        ('A', 'foo', 1010): ['1.2.3.4', '5.5.4.5'],
+        ('AAAA', '*', 100023): ['::1', '::2', '::3', '::4'],
+    }
+
+    ADDITIONAL_TEST_DATA = {
+        ('A', 'zekdi', 99): ['134.48.204.28', '151.85.162.150', '5.174.133.123', '96.37.218.195', '106.18.66.163',
+                             '51.75.149.213', '9.105.0.185', '32.198.60.88', '93.141.131.151', '6.133.10.124'],
+        ('A', 'knebq', 82): ['218.154.60.184'],
+    }
+
+    @classmethod
+    def _create_rr_sets(cls, data, domain):
+        rr_sets = []
+        rrs = {}
+        for (type_, subname, ttl), rr_contents in data.items():
+            rr_set = RRset(domain=domain, subname=subname, type=type_, ttl=ttl)
+            rr_sets.append(rr_set)
+            rrs[(type_, subname)] = this_rrs = []
+            rr_set.save()
+            for content in rr_contents:
+                rr = RR(content=content, rrset=rr_set)
+                this_rrs.append(rr)
+                rr.save()
+        return rr_sets, rrs
+
+    @classmethod
+    def setUpTestDataWithPdns(cls):
+        super().setUpTestDataWithPdns()
+        cls.rr_sets, cls.rrs = cls._create_rr_sets(cls.TEST_DATA, cls.full_domain)
+
+    def test_empty_domain_create_single_empty(self):
+        with PDNSChangeTracker():
+            RRset.objects.create(domain=self.empty_domain, subname='', ttl=60, type='A')
```

Comment:
We probably should also test if we can successfully create records as a keyword argument to the `RRset` constructor (i.e. a test similar to this one, that has `records=["1.2.3.4","5.6.7.8"]` added). Or do we not want to support that? (Not sure if it worked before.)

Reply:
We can certainly add that if there is a use-case for it (depends on how we implement the serializers)

Reply:
actually, as your bringing up whether there's a use-case... maybe not. I'm ok with skipping `records=` in the `RRset` constructor. Probably a bunch of work for little benefit

Reply:
For tests, I already discovered quite some use-cases